src/jit/morph.cpp

```diff
@@ -9550,8 +9550,7 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
         {
             GenTreePtr effOp1 = op1->gtEffectiveVal();
             
-            noway_assert( (effOp1->gtOper == GT_CNS_INT) &&
-                         ((effOp1->gtIntCon.gtIconVal == 0) || (effOp1->gtIntCon.gtIconVal == 1)) );
+            noway_assert(effOp1->IsZero() || effOp1->IsIOne());
```

Comment:
Besides I don't feel comfortable with liberal checks and depending on the context for those checks to be correct for two reasons: 1) it could introduce hard to diagnose bugs 2) for a person trying to read the code it gives wrong understanding of the code potentially unless one is very careful.